reproducibility/codalab/reproduce_results.py

```diff
@@ -0,0 +1,436 @@
+                                f"wilds:{wilds_src_uuid}",
+                                f"{dataset_fullname}:{dataset_uuid}",
+                                self._construct_command(
+                                    dataset,
+                                    algorithm=algorithm,
+                                    seed=seed,
+                                    hyperparameters={},
+                                ),
+                            ]
+                        )
+            # TODO: ADD one offs -Tony: GroupDro and ID
+
+    def analyze(self, in_distribution=False):
+        worksheet_uuid = self._set_worksheet()
+        datasets = (
+            dataset_defaults.keys()
+            if not in_distribution
+            else list(ID_HYPERPARAMETER_SEARCH_SPACE["datasets"].keys())
+        )
+
+        for dataset in datasets:
+            if (
+                not self._all_datasets
+                and dataset in CodaLabReproducibility._ADDITIONAL_DATASETS
+            ):
+                continue
+
+            dataset_results = dict()
+            algorithms = (
+                [key for key in algorithm_defaults.keys()]
+                + CodaLabReproducibility._CIVIL_COMMENTS_ADDITIONAL_ALGORITHMS
+                if dataset == "civilcomments"
+                else algorithm_defaults.keys()
+            )
+            for algorithm in algorithms:
+                dataset_results[algorithm] = []
+
+                # Skip the additional datasets if we don't want to reproduce results for all datasets
+                if (
+                    (algorithm in ["IRM", "deepCORAL"] and dataset == "civilcomments")
+                    or algorithm == "groupDRO"
+                    or (in_distribution and algorithm != "ERM")
+                ):
+                    continue
+
+                if dataset == "poverty":
+                    folds = ["A", "B", "C", "D", "E"]
+                    for fold in folds:
+                        bundle_name = (
+                            f"{dataset}_{algorithm}_fold{fold}"
+                            if not in_distribution
+                            else f"hp_id_{algorithm}_{dataset}_fold{fold}"
+                        )
+                        uuid = self._get_bundle_uuid(
+                            bundle_name, worksheet_uuid
+                        )
+                        results_dfs = load_results(
+                            f"https://worksheets.codalab.org/rest/bundles/{uuid}/contents/blob",
+                            splits=["val", "test"],
+                            include_in_distribution=True,
+                        )
+                        dataset_results[algorithm].append(results_dfs)
+                else:
+                    # 10 different seeds were used for camelyon17. 3 different seeds for other datasets.
+                    seeds = range(0, 10) if dataset == "camelyon17" else [0, 1, 2]
```

Comment:
Since this special-casing + the folds for poverty keep coming up, what do you think of factoring them out? It's a bit clunky since fold is different from seed... but maybe in the config, each dataset can store the keyword (`seed` or `fold`) and also whether they should loop through `range(2)` or `range(10)` or `['A', 'B', 'C', 'D', 'E']`?

Reply:
That makes sense. I will update before merging this PR.